ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13456,6 +13457,26 @@ ASTNode analyzeCreateTable(
       }
     }
 
+    HiveStorageHandler handler = null;
+    try {
+      handler = HiveUtils.getStorageHandler(conf, storageFormat.getStorageHandler());
+    } catch (HiveException e) {
+      throw new SemanticException("Failed to load storage handler:  " + e.getMessage());
+    }
+
+    if (handler != null) {
+      String fileFormatPropertyKey = handler.getFileFormatPropertyKey();
+      if (fileFormatPropertyKey != null) {
+        if (tblProps != null && tblProps.containsKey(fileFormatPropertyKey) && storageFormat.getSerdeProps() != null &&
```

Comment:
A quick clarifying question: this section handles the case when both `STORED AS ORC` and `TBLPROPERTIES` are defined in the same DDL query? If so, why do we need the `&& storageFormat.getSerdeProps() != null` part? If we used `||` instead, would that work too for validating both the tblproperties and the serdeproperties case?

Reply:
In `StorageFormat` class I store the fileformat in the serdeproperties, because I do not have reference to the table properties (it might be the case that the tableproperties were not parsed yet)